File: app/src/test/java/de/jodamob/kotlin/droidcon/e/GreeterTest.java
package de.jodamob.kotlin.droidcon.e;

import org.junit.Test;

import java.util.List;

import static junit.framework.Assert.assertEquals;
import static junit.framework.Assert.assertTrue;

public class GreeterTest {

    Greeter tested = new Greeter();

    @Test
    public void should_greet() {
        assertEquals("Hello Droidcon", tested.greet("Droidcon"));
    }

    @Test
    public void should_be_interntional() {
        List<String> greetings = tested.getSomeGreetings();
        assertTrue(greetings.contains("Hallo"));
    }
}

File: app/src/test/java/de/jodamob/kotlin/droidcon/e/GreeterTest.kt
package de.jodamob.kotlin.droidcon.e

import org.amshove.kluent.`should contain`
import org.amshove.kluent.`should equal`
import org.junit.Test
import kotlin.reflect.KFunction1

class GreeterTest {

    val tested = Greeter()

    @Test
    fun `should greet the conference`() {
        calling {tested::greet} with "Droidcon" `should equal` "Hello Droidcon"
    }

    fun calling(function: () -> KFunction1<String, String>) = function

    infix fun (() -> KFunction1<String, String>).with(argument : String) : String =
        invoke().invoke(argument)

    @Test
    fun should_be_interntional() {
        val greetings = tested.someGreetings
        greetings `should contain` "Hallo"
    }
}

